botfront/imports/lib/story.utils.tests.js

```diff
@@ -376,3 +552,36 @@ describe('proper flattening of stories', function() {
         expect(flattenStory(checkpointedStory)).to.be.deep.equal(flattenedStory);
     });
 });
+
+if (Meteor.isServer) {
+    describe('getAllTemplates', () => {
+        it('fetch English responses', async () => {
+            const response = await getAllTemplates('test', 'en');
+            expect(response).to.be.deep.equal(
+                responsesExported.en,
+            );
+        });
+        it('fetch ZZ responses', async () => {
+            const response = await getAllTemplates('test', 'zz');
+            expect(response).to.be.deep.equal(
+                responsesExported.zz,
+            );
+        });
+        it('fetch English and ZZ responses', async () => {
+            const response = await getAllTemplates('test', ['en', 'zz']);
+            expect(response).to.be.deep.equal(
+                responsesExported['en-zz'],
+            );
+        });
+        it('fetch all responses', async () => {
+            const response = await getAllTemplates('test');
+            expect(response).to.be.deep.equal(
+                responsesExported['en-zz'],
+            );
+        });
+        it('fetch no responses', async () => {
+            const response = await getAllTemplates('test', 'nonexisting');
+            expect(response).to.be.deep.equal({});
+        });
+    });
+}
```

Comment:
I think important test cases are missing:
- Testing that the NLG returns the expected response in the expected language
- Testing that the domain is generated as expected

Reply:
/* to do:
extractDomain
getStoriesAndDomain
accumulateExceptions
*/

These were never actually covered, all I did is add coverage for getAllTemplates.

Re: Testing that the NLG returns the expected response in the expected language: Yes, we should have tests for our graphql routes.